src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -136,6 +136,56 @@ case class CsvRelation protected[spark] (
     }
   }
 
+  override def buildScan(requiredColumns: Array[String]): RDD[Row] = {
```

Comment:
Would you please add some comments to this method and explain the general logic? This will help other developers quickly understand what is going on when they want to contribute to this code.